src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -0,0 +1,135 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab expandtab
+
+#include "crimson/os/seastore/extent_placement_manager.h"
+#include "crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.h"
+#include "crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h"
+#include "crimson/os/seastore/onode_manager/staged-fltree/node_extent_manager/seastore.h"
+#include "crimson/os/seastore/collection_manager/collection_flat_node.h"
+#include "crimson/os/seastore/object_data_handler.h"
+
+namespace crimson::os::seastore {
+
+ExtentAllocWriter::write_ertr::future<>
+ExtentAllocWriter::write(CachedExtent* extent) {
+  bufferlist bl;
+  extent->prepare_write();
+  bl.append(extent->get_bptr());
+  auto segment = extent->get_allocated_segment();
+  assert(extent->extent_writer);
+  assert(segment);
+  auto old_segment_iter = open_segments.begin();
+  if (segment.get() != old_segment_iter->get()) {
+    // segment writes are ordered, so when an opened segment isn't the current
+    // one being written to, it is full and should be closed.
+    (void) open_segments.front()->close();
```

Comment:
do we have the situation cross the segments  to save data or just ignore the tail space when it is not enough?

Reply:
i think we should go with the latter approach for the sake of simplicity.